# 36) Bernoulli Dağılımı
Bernoulli dağılımı, yalnızca iki olası sonucu (başarı/başarısızlık, evet/hayır veya 1/0) olan tek bir denemenin olasılığını inceleyen en basit kesikli olasılık dağılımıdır. Başarı olasılığı p ile, başarısızlık olasılığı ise q = 1-p ile gösterilir.

Temel Özellikler ve Formülü
1. Başarı durumu (X = 1) olasılığı: p
2. Başarısızlık durumu (X = 0) olasılığı: 1 - p
3. Tek parametresi vardır: p (0 ≤ p ≤ 1)
4. Beklenen değer (ortalama): E(X) = p
5. Varyans: Var(X) = p(1 - p)


### Bernoulli Dağılımına bazı Örnekler
Kullanım Alanları
- Yazı-Tura Atma: Bir paranın yazı (1) veya tura (0) gelmesi.
- Kalite Kontrol: Üretilen bir ürünün sağlam (1) veya kusurlu (0) çıkması.
- Ders Geçme Durumu: Bir öğrencinin dersten geçmesi (1) veya dersten kalması.

Bernoulli dağılımı, tek bir denemeyi modelliyor. Bu tek denemeyi **n kere tekrarlarsak** (örneğin parayı 10 kere atarsak), elde ettiğimiz "toplam başarı sayısı" dağılımına **Binom Dağılımı** denir (Konu 40).

In [3]:
import numpy as np
from scipy import stats

# Bernoulli dağılımının tek parametresi: başarı olasılığı (p)
p = 0.3  # Örnek: bir reklamın tıklanma olasılığı %30 olsun

# --- Teorik Değerler ---
# Bernoulli dağılımının formülünden E(X) ve Var(X) hesaplıyoruz.
teorik_beklenen_deger = p
teorik_varyans = p * (1 - p)

print(f"Teorik Beklenen Değer E(X): {teorik_beklenen_deger}")
print(f"Teorik Varyans Var(X): {teorik_varyans}")

# --- Simülasyon (Deneysel Değerler) ---
# scipy'nin bernoulli.rvs() fonksiyonu, p olasılıklı 10000 tane 
# rastgele Bernoulli denemesi (0 veya 1) üretir
np.random.seed(42)
denemeler = stats.bernoulli.rvs(p, size=10000)

deneysel_ortalama = np.mean(denemeler)
deneysel_varyans = np.var(denemeler)

print(f"\nDeneysel Ortalama (10000 deneme): {deneysel_ortalama}")
print(f"Deneysel Varyans (10000 deneme): {deneysel_varyans}")

# --- Karşılaştırma ---
print(f"\nOrtalama farkı: {abs(teorik_beklenen_deger - deneysel_ortalama):.4f}")
print(f"Varyans farkı: {abs(teorik_varyans - deneysel_varyans):.4f}")

print(f"1'lerin sayısı: {np.sum(denemeler == 1)}")
print(f"0'ların sayısı: {np.sum(denemeler == 0)}")

Teorik Beklenen Değer E(X): 0.3
Teorik Varyans Var(X): 0.21

Deneysel Ortalama (10000 deneme): 0.2887
Deneysel Varyans (10000 deneme): 0.20535231000000004

Ortalama farkı: 0.0113
Varyans farkı: 0.0046
1'lerin sayısı: 2887
0'ların sayısı: 7113


### Sonuç
Teorik değerler (E(X)=0.3, Var(X)=0.21) ile 10000 denemelik simülasyondan elde ettiğimiz deneysel değerler (ortalama=0.2887, varyans=0.2054) birbirine oldukça yakın çıktı. Bu yakınlık, Bernoulli dağılımının formüllerinin gerçek veriyle tutarlı olduğunu doğruluyor. Deneme sayısını 10000'den 100000'e çıkarsaydık, Büyük Sayılar Yasası gereği şans faktörü daha da azalacağı için bu farkın şu andaki farktan (0.0113) daha da küçük çıkmasını beklerdik.